### DISCUSSION TITLE
How to build for C++ development?

### DISCUSSION THREAD

**Discussion by Matthew-w56** [rism-digital/verovio#3604] [Q&A] [answered]
Hello!

I just found this library as I work to build a sheet music editing software.  I think this library is awesome, and have tested it out in JavaScript.  However, I am embedding the library in a Flutter project and C++ fits my needs better for that.  But I can't get it to build in C++.  Any tips?

I notice that the page that shows how to build is meant for the command line application, which I can get to build just fine.  But what should I change from those instructions in order to code with it in C++?

Thanks!

Update:
I used the -DBUILD_AS_LIBRARY=ON flag and got the .dll and then used "nmake install" to install the dll and all the Verovio/share/verovio files into my Program Files (x86) folder. However, it never installed any headers, and I have been unable to link it up without them. I have also tried manually copying over the contents of the tools/include folder as well but that didn't help too much. I'm quite confused here. Any help or step-by-steps would save me here

Note:
I don't necessarily need to be able to work with the library IN C++. My application will be with the Flutter framework, which uses Dart. Importing the library into this framework is possible through creating a FFI in Dart for a C library. So ideally, I would like to get the C wrapper version of the library, so I can export it into my project and begin working with it that way.

Thanks for any pointers and help!

**Comment by lpugin** [comment, marked as answer]
I adjusted cmake to copy header files. See [this](https://github.com/rism-digital/verovio/commit/f8f4c113aae629de9071e69ae4c3edbcb0f051b4) commit. I seems to work on macOS. Could you try on Windows?

**Comment by Matthew-w56** [comment]
I just tested it out and it works great, thanks!  The only other question I would have for that build is that, even when building for 64 bit, it still defaults to Program Files (x86).  Is there something that can be added to have it install to Program Files when the dll is being built by 64 bit tools?

**Reply by lpugin** [reply]
I added some documentation in the [reference book](https://book.verovio.org/installing-or-building-from-sources/library.html)
